dandiapi/api/tasks/scheduled.py

```diff
@@ -127,7 +128,17 @@ def refresh_materialized_view_search() -> None:
     updated without locking the table.
     """
     with connection.cursor() as cursor:
+        # This query may take a long time, so we explicitly set a timeout with
+        # postgres that is slightly less than the Celery soft_time_limit to
+        # avoid it running too long.
+        # Note that setting Celery time limits alone is not sufficient. If the task times out,
+        # Celery will stop the Python execution, but not the DB query.
+        cursor.execute('BEGIN;')
+        cursor.execute(
+            'SET LOCAL statement_timeout = %s;', [(REFRESH_MATERIALIZED_VIEW_TIMEOUT * 1000) - 5]
```

Comment:
Just so you know this is setting the timeout to 5 milliseconds shorter than `REFRESH_MATERIALIZED_VIEW_TIMEOUT`, not 5 seconds. Not sure if that's what you intended.

Reply:
Good catch. It was not intended, I was bouncing between a couple PRs and did it quickly without thinking enough :sweat_smile: